include/zephyr/data/navigation.h

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2023 Trackunit Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_DATA_NAVIGATION_H_
+#define ZEPHYR_INCLUDE_DATA_NAVIGATION_H_
+
+#include <zephyr/types.h>
+
+/**
+ * @brief Navigation utilities
+ * @defgroup navigation Navigation
+ * @ingroup utilities
+ * @{
+ */
+
+/**
+ * @brief Navigation data structure
+ *
+ * @details The structure describes the momentary navigation details of a
+ * point relative to a sphere (commonly Earth)
+ */
+struct navigation_data {
+	/** Latitudal position in nanodegrees (0 to +-180000000000) */
+	int64_t latitude;
+	/** Longitudal position in nanodegrees (0 to +-180000000000) */
+	int64_t longitude;
+	/** Bearing angle in millidegrees (0 to 360000) */
+	uint32_t bearing;
+	/** Speed in millimeters per second */
+	uint32_t speed;
+	/** Altitude in millimeters */
+	int32_t altitude;
+};
+
+/**
+ * @brief Calculate the distance between two navigation points along the
+ * surface of the sphere they are relative to.
+ *
+ * @param distance Destination for calculated distance in millimeters
+ * @param p1 First navigation point
+ * @param p2 Second navigation point
+ *
+ * @return 0 if successful, -EINVAL if any navigation point is invalid
```

Comment:
use multiple `@retval` instead
also, any-->either?

Reply:
Using multiple @return, and changed any to either